### ISSUE TITLE
Multi-region lambda configurations

### ISSUE THREAD

**Issue by tjrivera** [Kong/kong-plugin-aws-lambda#47]
Hi Folks, 

I have some internal teams that have begun leveraging this plugin. One use-case that we've encountered, is that teams will deploy the same lambda function into multiple regions for an active-active setup. 

For context, on the gateway side, we have instances that run in multiple regions, we recommend teams configure non-lambda-based services to the gateway with a latency-based DNS record in their configuration so that an incoming request that hits our west region, would then proxy to an upstream in west, vice versa for east, or any other region. 

With lambdas we can't rely on latency-based DNS routing, and our configuration is global. This means teams can only configure a single lambda per-service. 

We've thought about some workarounds that involve configuring multiple services per-region and then a global service that refers to those Kong services, but this isn't ideal. 

I imagine it may be a great deal of work to passively healthcheck configured lambdas to enable failover to another arn, but I wanted to get this out there to get folks thoughts on a possible solution. In this case our primary goal is availability and not necessarily improved latency via geolocation.

**Comment by Tieske** [comment]
building healthchecks into the plugin is a no-go to me. But an option would be to expose `upstreams` to plugins, such that they can use those for DNS resolution, which then automatically would enable all other properties for the plugin as well (which includes healthchecks).

But in this case, the AWSv4 signature for the request will include the host name, so rerouting upon failure would require resigning the request probably.

fyi @locao @hbagdi wrt balancer features and use cases

**Comment by tjrivera** [comment]
> building healthchecks into the plugin is a no-go to me. But an option would be to expose upstreams to plugins, such that they can use those for DNS resolution, which then automatically would enable all other properties for the plugin as well (which includes healthchecks).

Makes sense. If this is feasible I think it could be a powerful pattern but I'm not clear on the work involved. It feels like it would be a change to core Kong functionality and patterns.
